packages/core/src/filters/FilterSystem.ts

```diff
@@ -170,17 +173,28 @@ export class FilterSystem extends System
         state.legacy = legacy;
 
         state.target = target;
-
         state.sourceFrame.copyFrom(target.filterArea || target.getBounds(true));
 
         state.sourceFrame.pad(padding);
+
         if (autoFit)
         {
-            state.sourceFrame.fit(this.renderer.renderTexture.sourceFrame);
+            const sourceFrameProjected = this.renderer.renderTexture.sourceFrame.clone();
+
+            // Project source frame into world space (if projection is applied)
+            this.transformAABB(renderer.projection?.transform?.clone().invert() || Matrix.IDENTITY, sourceFrameProjected);
```

Comment:
creating a new matrix each time here! could store the invert?

Reply:
Done 👍